internal/connect/product.go

```diff
@@ -115,3 +129,27 @@ func (p Product) distroTarget() string {
 	version := strings.Split(p.Version, ".")[0]
 	return identifier + "-" + version + "-" + p.Arch
 }
+
+// SCC and RMT servers send the ID as an int.
+// But SMT sends it as a string in POST migrations.
+// Try to decode as an int first, then as a string.
+func findID(data []byte) int {
+	idInt := struct {
+		ID int `json:"id"`
+	}{}
+	if err := json.Unmarshal(data, &idInt); err == nil {
+		return idInt.ID
+	}
+
+	// Decoding id as an int failed. Try to decode as a string.
+	idStr := struct {
+		ID string `json:"id"`
```

Comment:
can you do `ID int json:"id,string"` here and drop manual conversion below?

Reply:
yep - that works - nice!

Reply:
another idea: make custom unmarshaller only for the ID field like here: https://go.dev/play/p/rityF2vTsFp?v=goprev.go?download=true

Reply:
Thanks, I'll give that a try in another branch and see what it looks like. It's shorter, but adds a new type. Not sure which way is best.

Reply:
> another idea: make custom unmarshaller only for the ID field like here: https://go.dev/play/p/rityF2vTsFp?v=goprev.go?download=true

https://github.com/djoreilly/connect-ng/commit/f742a5ab55046869af00918b633b3009c6c6b0ff